include/linker/linker-defs.h

```diff
@@ -172,9 +172,15 @@ GDATA(__data_num_words)
 
 #include <zephyr/types.h>
 /*
- * The following are externs symbols from the linker. This enables
+ * Memory owned by the kernel, to be used as shared memory between
+ * application threads.
+ *
+ * The following are extern symbols from the linker. This enables
  * the dynamic k_mem_domain and k_mem_partition creation and alignment
  * to the section produced in the linker.
+
+ * The policy for this memory will be to initially configure all of it as
+ * kernel / supervisor thread accessible.
```

Comment:
No complaints with the docs, but one thing that would be good in the future would be to abstract out all the handling for these __*_start/end symbols (we use them a lot), which would obviate the text here and replace it with something like a DEFINE_BOUNDED_SECTION() macro instead.  No current plans, but I hope it happens someday.

Reply:
that would be nice, indeed.